### DISCUSSION TITLE
How does google or tools select vehicle for routing

### DISCUSSION THREAD

**Discussion by Gyantoshkumar** [google/or-tools#2726] [Routing (and legacy CP) questions]
Hi ! does  google or tools always do fastest delivery for vehicle routing . 

I mean does google or tools select vehicle randomly or it tries to distribute the orders equally ?  

is it possible to implement google or tools for cumulative shortest distance/time  for all vehicle

**Comment by Mizux** [comment]
By default routing solver will  try to minimize the cumulative shortest distance/time for all vehicle, which often mean use a number of vehicle as minimum as possible. i.e. usually you start by having only the sum of arc cost visited as your objective value to minimize...

Otherwise VRP is NP hard so finding and proving optimality is not always doable but solver will try hard to find it using various heuristic and Local Neighborhood Search.
